src/wallet/wallet.cpp

```diff
@@ -2363,10 +2363,10 @@ void CWallet::AvailableCoins(std::vector<COutput> &vCoins, bool fOnlySafe, const
                 continue;
             }
 
-            bool fSpendableIn = ((mine & ISMINE_SPENDABLE) != ISMINE_NO) || (coinControl && coinControl->fAllowWatchOnly && (mine & ISMINE_WATCH_SOLVABLE) != ISMINE_NO);
-            bool fSolvableIn = (mine & (ISMINE_SPENDABLE | ISMINE_WATCH_SOLVABLE)) != ISMINE_NO;
+            bool solvable = IsSolvable(*this, pcoin->tx->vout[i].scriptPubKey);
+            bool spendable = ((mine & ISMINE_SPENDABLE) != ISMINE_NO) || (coinControl && coinControl->fAllowWatchOnly && solvable);
```

Comment:
Imo it'd improve readability here to add a check to the second predicate here:
```c++
((mine & ISMINE_WATCH_ONLY) != ISMINE_NONE) && ...
```
While redundant, it's not immediately obvious that invalid watch-only's were filtered out a few lines up. It would also guard against future oopses if a new type is ever added.

Reply:
That would be incorrect. When `(coinControl && coinControl->fAllowWatchOnly)` we're expected to set `spendable = true` in `COutPut` for `ISMINE_WATCH_ONLY` outputs.

Reply:
Oh, nevermind. I misread the `!=`. Will do.